### ISSUE TITLE
Visual Studio Code Freezes all the time with many search results

### ISSUE THREAD

**Issue by ranbuch** [Microsoft/vscode#30403]
- VSCode Version: Code 1.13.1 (379d2efb5539b09112c793d3d9a413017d736f89, 2017-06-14T18:21:47.485Z)
- OS Version: Windows_NT ia32 10.0.15063
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|stylus|buzinas|0.1.0|
|gltf-vscode|cesium|2.0.3|
|csharp|ms-vscode|1.11.0|
|vscode-docker|PeterJausovec|0.0.16|
|cordova-tools|vsmobile|1.2.7|;

---

Steps to Reproduce:

1.Open VS-Code
2.Start working

If you want it to snap out of it you need to press ```Alt``` + ```Tab``` and then you'll see the cursor blinking until you'll return to focus on the VS-Code window.

**Comment by ShiftySituation** [comment]
Yes... the new build isn't working.   Falling in line with other failed Microsoft products.  Keep up the good work?

**Comment by rebornix** [comment]
@ranbuch do you also see this issue when you run VSCode without any extension by `code --disable-extensions` ?

**Comment by luchillo17** [comment]
@rebornix Freezes at start, but works without extensions, just tried and it doesn't happen with `--disable-extensions`.

My ENV:
OSX: Sierra 10.12.5
Node: 7.10.0
NPM: 5.0.3

**Comment by luchillo17** [comment]
@rebornix I want to disable all extensions and try and activate one at a time to see where it crashes, but it doesn't let me disable any with `--disable-extensions`, any idea how to?

**Comment by aervin** [comment]
Running VSCode with --disable-extensions seemed to help. VSCode processes no longer grabbing up all available CPU 75% of the time.

**Comment by luchillo17** [comment]
Temporal fix downloaded the 1.13.1 again and disabled updates.
That's ugly.

**Comment by chrmarti** [comment]
@luchillo17 Could you list your extensions with `code --list-extensions` and append here?

**Comment by PooperPig** [comment]
I'm having very similar experience developing angular 2 with the following extensions
Angular.ng-template
christian-kohler.path-intellisense
CoenraadS.bracket-pair-colorizer
EditorConfig.EditorConfig
eg2.tslint
GregorBiswanger.json2ts
johnpapa.angular-essentials
johnpapa.Angular2
msjsdiag.debugger-for-chrome
natewallace.angular2-inline
PKief.material-icon-theme
rbbit.typescript-hero

I will try to identify culprits by a process of elimination & report back if successful.

**Comment by chrmarti** [comment]
The immediate freeze at startup could be #30407 that is triggered by some snippets in extensions.

The problem reported by @ranbuch seems to be a different problem. @ranbuch Does the problem reproduce if you launch with `code --disable-extensions`?

**Comment by PooperPig** [comment]
It appears that Typescript Hero may be the culprit in my case.
With it disabled performance seems normal - with it enabled VS Code shows high CPU usage and is very slow to respond.
I notice this wasn't in the OP's list of add-ons, but maybe it will help with others, and may help identify the root cause.

note: I only installed TS Hero a few days ago whereas the other extensions I've been using for a while - so it's possible that TS Hero has this problem anyway, and it is unrelated to the current new release of VS Code - just that I hadn't experienced it previously.

**Comment by luchillo17** [comment]
@chrmarti My issue is not performance related, i don't see my CPU usage going up that hard, it just freezes the moment i open the app, then later it ask me if i want to keep waiting or close the app.

As for the list of plugins, i have many, at least 15 i think, let me see if i find the list the Sync plugin generates in github...

Ok here we go:
https://pastebin.com/FA8iiHHG

**Comment by chrmarti** [comment]
@PooperPig Thanks, that helps!

@luchillo17 You have the Rails snippets installed, these are known to trigger #30407. Could you try running `code --uninstall-extension atefth.ruby-on-rails-snippets` (or use the UI if you're back on 1.13) and then launch code 1.14 to see if that works around the problem?

**Comment by luchillo17** [comment]
@chrmarti Indeed it works now, however does it means i cannot use such plugin until either they patch it or VSCode does?

**Comment by chrmarti** [comment]
@luchillo17 Correct, the fix for that problem (#30407) is lined up for the recovery build (1.14.1). We have no ETA yet, but I think it's less than a week.

Keeping this issue open for the originally reported problem against 1.13.1.

**Comment by ranbuch** [comment]
When disabling all extensions works good.
```
ms-vscode.csharp
vsmobile.cordova-tools
PeterJausovec.vscode-docker
cesium.gltf-vscode
buzinas.stylus
```
Those are all my extensions.

I think you should close this one.

Thank you!

**Comment by ranbuch** [comment]
Just append again :(

**Comment by icyJoseph** [comment]
Hi!

I've been having the same issue, please refer to my comment here:

https://github.com/Microsoft/vscode/issues/28709

**Comment by chrmarti** [comment]
@ranbuch Can you confirm this still occurs with 1.14? (release yesterday)

**Comment by icyJoseph** [comment]
@chrmarti Sorry for meddling, in my case it still happened after the update.

**Comment by ShiftySituation** [comment]
@chrmarti I can... after the update yesterday 1.14 is when we started experiencing the issues.

**Comment by chrmarti** [comment]
@icyJoseph That sounds like another problem, could you open a separate issue for that?

**Comment by malagant** [comment]
I have the same trouble but since 1.14. Disabling the extensions helps, but how should I be productive without the extensions? ;)

Any hints on how to find out, wich extension is problematic? Disabling one after another?

**Comment by malagant** [comment]
Ok, I deleted ~/.vscode and installed the recommended extensions for my projects. Everything works again. Maybe, this will work for others, too.

**Comment by dwally89** [comment]
Any latest on this? Is it safe to install 1.14 (or a more recent version) yet?

**Comment by chrmarti** [comment]
@dwally89 Please try the latest 1.14.1 and open a new issue if you still see any problems. This issue is waiting on feedback from @ranbuch on whether 1.14.1 fixes the reported problem.

**Comment by ranbuch** [comment]
@vscodebot[bot] 

The reason may be related to the ```SEARCH``` menu?
If I've searched the hole project (I have a big project) for some statement and I've changed the code it refreshes the search results.

I think that if I'm pressing ```Clear Search Results``` the issue isn't reproducing.

**Comment by chrmarti** [comment]
Reopening for the perf issue with many search results.

**Comment by kevinw** [comment]
I'm still seeing this issue.

**Comment by roblourens** [comment]
Is it happening when there are a large number of search results too?

**Comment by kevinw** [comment]
ah, sorry, i didn't realize this issue was just for search results. it happens with zero search results, and with the search results not visible.

**Comment by roblourens** [comment]
Ok, you can probably open a new issue for that. There are different ways to see which process is eating cycles depending on the OS you're using.

**Comment by kevinw** [comment]
I can see in task manager that it is indeed VSCode taking the cycles--the problem is just figuring out why VSCode hangs. Can I profile VSCode easily by doing the extension debugger path?

**Comment by roblourens** [comment]
There are a bunch of VS Code processes, it depends on which one is causing the issue.

**Comment by dmt-13** [comment]
Same issue for me. It takes up to 2 minutes to find a text in files that are in the folder and also it freezes while searching. I tried to search in a freshly installed stable version - result is just the same. It's hard to use VSC without ability to search across the project..

- VSCode Version: Code - Insiders 1.17.0-insider (128a4e3640c5375f7b0bc1cfd37f626481b1386f, 2017-09-12T05:10:59.101Z)
- OS Version: Darwin x64 16.5.0
- Extensions: Extensions are disabled
---

**Comment by roblourens** [comment]
This covers a lot of ground and lots has changed - if anyone is having issues, please open a new issue with as many details as possible, also see https://github.com/Microsoft/vscode/wiki/Performance-Issues
